[DOCS] Add info about ILM and unallocated shards. (#56655)
elastic/elasticsearch: docs/reference/ilm/ilm-index-lifecycle.asciidoc

The minimum age defaults to zero, which causes {ilm-init} to move indices to the next phase
as soon as all actions in the current phase complete. 

> If an index has unallocated shards and the cluster is in the _yellow_ state, 

Reviewer: It may be helpful to disambiguate between this and the general cluster state.
```suggestion
If an index has unallocated shards and the <<cluster-health,cluster health status>> is yellow, 
```

> the index can still transition from one phase to the next according to its {ilm} policy, 
> but there might be unexpected side effects because {es} can only perform 
> certain clean up tasks on a healthy cluster. 

Reviewer: Some rewording suggestions here to shorten and change order. Feel free to ignore if wanted.
```suggestion
the index can still transition to the next phase according to its {ilm} policy.
However, because {es} can only perform certain clean up tasks on a green
cluster, there may be unexpected side effects. 
```

For example, if you have a policy that moves shards to warm nodes when 
the index transitions out of the hot phase, {es} won't delete shards that 

> weren't fully replicated from the hot nodes. 

Reviewer 2: This sentence is confusing for me, and I even know what's going on. To me it sounds mutually exclusive with the previous sentence, saying things like "we don't delete shards that haven't moved over" right after saying "there may be unexpected side effects"

> For example, if you have a policy that moves shards to warm nodes when 
> the index transitions out of the hot phase, {es} won't delete shards that 
> weren't fully replicated from the hot nodes. 

Reviewer: I took a stab at the suggestion, but feel free to ignore if I'm off base. 
I think this might benefit from being in its own para.
```suggestion

For example, when an index transitions from the hot phase to the warm phase,
your policy moves shards from hot nodes to warm ones. For a yellow cluster,
{es} won't clean up any shards on hot nodes that weren't fully replicated. For
a healthy green cluster, these shards would be deleted as part of cleanup.

```

Author: I think the example is just making this more complicated than it needs to be. The key takeaway is that the index will still move through the lifecycle, but bad things can happen.
